src/server/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -63,12 +82,27 @@ namespace ts.server.typingsInstaller {
                 this.log.writeLine(`Process id: ${process.pid}`);
             }
             this.npmPath = getNPMLocation(process.argv[0]);
-            this.exec = require("child_process").exec;
-            this.httpGet = require("http").get;
+            let execSync: ExecSync;
+            ({ exec: this.exec, execSync } = require("child_process"));
+
+            this.ensurePackageDirectoryExists(globalTypingsCacheLocation);
+
+            try {
+                if (this.log.isEnabled()) {
+                    this.log.writeLine(`Updating ${TypesRegistryPackageName} npm package...`);

```

Comment:
Note that I believe using `npm install` on an already installed package with loose versioning will not update it. You may want to use `npm update` (which will also install if not present). See http://stackoverflow.com/questions/12478679/npm-install-vs-update-whats-the-difference